ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,141 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiConvexECOSExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using Convex
+    using ECOS
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..Convex
+    using ..ECOS
+end
+
+# Use other necessary libraries
+using LinearAlgebra: eigvals
+const MOI = Convex.MOI
+
+# Use additional symbols that are not exported
+using Trixi: PolynomialOptimizer, @muladd
+
+# Import functions such that they can be extended with new methods
+import .PolynomialOptimizer: filter_eigvals, bisection, undo_normalization!
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+function stability_polynomials(cons_order, num_stage_evals, num_eig_vals,
+                               normalized_powered_eigvals_scaled, pnoms,
+                               gamma::Variable)
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
```

Comment:
This conversation was marked as resolved but I don't see an answer here in this thread?

Reply:
I marked this as resolved since I have gone with Michael's recommendation regarding avoiding the allocation inside the function by renaming `stability_polynomials!`, and etc. Is there anything else you would suggest me to do?

Reply:
Seems ok to me, thanks!